Update 99-openmediavault-dev-disk-by-id.rules

hannesd opened the pull request:
Hi,

I had the same problem described in https://github.com/openmediavault/openmediavault/issues/665. Because the USB device reported the same serial for two different USB hard drives the link under /dev/disks/by-id got overwritten by whichever drive was added last. In the front under Disks or SMART only one of the two drives was listed.

The already existing fix in /etc/udev/rules.d/99-openmediavault-dev-disk-by-id.rules does indeed correct the ID_SERIAL* env variables, but it will not fix the symlink and therefore the front end will only show one of the drives despite that rule.

I added the two SYMLINK lines. At first I added twice (used += instead of = twice). This, however, created a third symlink in .../by-id/. Two symlinks had the fixed ID_SERIAL and the third was the "old", faulty symlink. For reasons unclear to me it is that faulty symlink that gets picked up by openmediavault as the devicefile and while the command "omv-rpc DiskMgmt enumerateDevices" enumerates both drives correctly, the front end will still only show one drive (I assume that the front end does its own filtering based on the devicefile property).

So I ended up with the solution provided here: first overwrite any existing SYMLINKs and add the by-path and the fixed by-id symlinks.

However, I have a hunch that a better solution would be to rename the rules files to something < 60-...rules instead of 99-...rules. The file which creates the faulty symlink is /lib/udev/rules.d/60-persistent-storage.rules. If the ID_SERIAL field is fixed before that rules file is parsed the extra SMYLINK magic would not be required. Yet I cannot recommend this course of action as my knowledge of udev is very limited and I do not know what other unforeseen consequences this might have.

Cheers
-- Hannes


<!--
Thank you for opening a pull request! Here are some tips on creating a well formatted contribution.

Please give your pull request a title like "[Short description]"

This is the format for commit messages:

"""
[Short description]

[A longer multiline description]

Fixes: [ISSUE_URL or #ISSUE_ID, create one if necessary]

Signed-off-by: [YOUR_NAME] <[YOUR_EMAIL]>
"""

The Signed-off-by line is important, and it is your certification that your contributions satisfy the developers certificate or origin.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview. More information for contributors is available here:
https://docs.openmediavault.org/en/latest/development/contribute.html
-->

- [x] References issue
- [ ] Includes tests for new functionality or reproducer for bug

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/openmediavault/openmediavault?pullRequest=831) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/openmediavault/openmediavault?pullRequest=831) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/openmediavault/openmediavault?pullRequest=831) it.</sub>

votdev commented:
I will think about renaming the file to something like `60-...`to let do udev the SYMLINK magic. In the meanwhile i will merge your PR. Thx for the contribution.

votdev commented:
Could you please adapt the commit message according https://openmediavault.readthedocs.io/en/5.x/development/contribute.html#contribution-rules.

votdev commented:
Could you please test https://github.com/openmediavault/openmediavault/pull/835? I think renaming the file is the better solution than doing the SYMLINK magic ourself. Thanks for the tip.

hannesd commented:
My pleasure :)

Seeing how #835 does not work for me I have updated the commit message and signed the CLA.

votdev commented:
Thank you.

votdev commented:
Is it possible that you test https://github.com/openmediavault/openmediavault/pull/843 and check if it still works?

hannesd commented:
> Is it possible that you test #843 and check if it still works? The UDEV rules need to be more restrictive, so i added `ENV{DEVTYPE}=="disk", \`.

Works for me (JMicron case). I've set up a fresh 5.5.16-1 (Usul) install and replaced the file (`99-openmediavault-dev-disk-by-id.rules`) with that from the pull request. After a restart the symlinks in `/dev/disk/by-id/` are correctly named by our custom rule and all drives are listed in the GUI.

votdev commented:
> > Is it possible that you test #843 and check if it still works? The UDEV rules need to be more restrictive, so i added `ENV{DEVTYPE}=="disk", \`.
> 
> Works for me (JMicron case). I've set up a fresh 5.5.16-1 (Usul) install and replaced the file (`99-openmediavault-dev-disk-by-id.rules`) with that from the pull request. After a restart the symlinks in `/dev/disk/by-id/` are correctly named by our custom rule and all drives are listed in the GUI.

Have you tested the version that starts with
```
ACTION=="remove", GOTO="omv_dev_disk_by_id_end"
KERNEL!="sd*", GOTO="omv_dev_disk_by_id_end"
SUBSYSTEM!="block", GOTO="omv_dev_disk_by_id_end"
ENV{DEVTYPE}!="disk", GOTO="omv_dev_disk_by_id_end"
```
I'm asking because i have forced push a newer version at the same time you posted this comment.

hannesd commented:
> I'm asking because i have forced push a newer version at the same time you posted this comment.

It appears I have taken the code after you used the force. My file starts with the lines you mention and I just diffed the current state of https://github.com/openmediavault/openmediavault/pull/843/files vs. what's on my machine and the only thing that has changed are some comments.
